src/v/pandaproxy/json/requests/partition_offsets.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "json/json.h"
+#include "model/fundamental.h"
+#include "pandaproxy/json/types.h"
+#include "seastarx.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <rapidjson/encodings.h>
+
+namespace pandaproxy::json {
+
+struct topic_partition_offset {
+    model::topic topic;
+    model::partition_id partition;
+    model::offset offset;
+    bool operator<(const topic_partition_offset& other) const {
+        return std::tie(topic, partition)
+               < std::tie(other.topic, other.partition);
+    };
+};
+
+template<typename Encoding = rapidjson::UTF8<>>
+class partition_offsets_request_handler {
```

Comment:
i've seen this in a few places. should we have a supertype, mark all derived classes final & only override the methods that we want to return  data for ? 

the linker should get rid of any inheritance, specially if the super type has no data, it should get eliminated, but we would save on a bunch of 

```
    using Ch = typename Encoding::Ch;
    using rjson_parse_result = std::vector<topic_partition_offset>;
    rjson_parse_result result;

    bool Null() { return false; }
    bool Bool(bool) { return false; }
    bool Int64(int64_t) { return false; }
    bool Uint64(uint64_t) { return false; }
    bool Double(double) { return false; }
    bool RawNumber(const Ch*, rapidjson::SizeType, bool) { return false; }


```

redefinitions.

Reply:
> i've seen this in a few places. should we have a supertype, mark all derived classes final & only override the methods that we want to return data for ?

@senior7515 I agree. Thanks for keeping me honest.

Reply:
I already have some serialisation refactoring to do, so I'll leave it until then.